### DISCUSSION TITLE
upload hook and file check

### DISCUSSION THREAD

**Discussion by dani** [drakkan/sftpgo#879] [General]
Hi. I'm trying to add some controls on files which get uploaded (using FTP or SFTP, by unknown third parties). Most notably, I'd like to add ClamAV scanning, and reject files bigger than some threshold. I cannot use the pre-upload action, as at this point, nor the file size nor its content is available. I could use the upload action, but even if listed in execute_sync, and the hook replies with a 403 (it's an HTTP URL), the file is accepted, and acked to the client. Is there a way to execute an action just after the file is uploaded, check the file is OK, and if it's not, reject it with a notification to the client (a permission denied error) ?

**Comment by drakkan** [comment]
Hi,

the upload size limit is supported on a per-user basis, but currently there is no way reject a file from the upload hook. You can delete the file but no error is returned to the SFTP/FTP client.
Maybe I could consider adding something like this, are you interested in sponsoring this feature? Thank you

**Reply by dani** [reply]
I'd need to see this with my employer. Will let you know.

**Reply by drakkan** [reply]
> I'd need to see this with my employer. Will let you know.

#880 

however if everyone takes and no one returns the project will not be sustainable in the long run :cry:

**Reply by dani** [reply]
I'm very grateful :-)
I perfectly understand your need to make this project sustainable, and I'm quite sure the company I work in is willing to provide some support (we're currently evaluating SFTPGo and it could besome a key compnent in our infra). I'll speak to my superior tomorrow about a possible sponsorship.
